### PR TITLE
boards: stm32f411e_disco: Add pwm to drive green led

### PR DESCRIPTION
**Pull request by susundberg**
Dear all!

I started to look into the project, and i could not find any example of the Stm32F4 family having pwm led example working. 

With these changes my stm32f411e-disco board green led now fades with the samples/basic/fade_led. 
 
I hope i got it right, i looked for example from board/nucleo_g431rb.
 
Signed-off-by: Pauli Salmenrinne <susundberg@gmail.com>

### CODE REVIEW

File: boards/arm/stm32f411e_disco/pinmux.c [hunk 1]
```diff
@@ -18,6 +18,12 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PA2, STM32F4_PINMUX_FUNC_PA2_USART2_TX},
 	{STM32_PIN_PA3, STM32F4_PINMUX_FUNC_PA3_USART2_RX},
 #endif	/* CONFIG_UART_2 */
+
+
+
```
Comment (erwango) [inline boards/arm/stm32f411e_disco/pinmux.c]: Remove these lines

File: boards/arm/stm32f411e_disco/pinmux.c [hunk 2]
```diff
@@ -18,6 +18,12 @@ static const struct pin_config pinconf[] = {
...
+#ifdef CONFIG_PWM_STM32_4
+	{STM32_PIN_PD12, (STM32_PINMUX_ALT_FUNC_2 | STM32_PUSHPULL_NOPULL) },
```
Comment (erwango) [inline boards/arm/stm32f411e_disco/pinmux.c]: Please use dedicated helper define `STM32F4X_PINMUX_FUNC_FOO`

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this addition.
Few comments, also please update boards docs.

**Comment by susundberg** [comment]
TY for the comments. The first two i think were clear, but i am unsure about the board documentation update note. 

So i added to index.rst a very short "(pwm4 ch1)" comment. Please if this was not what was wanted can you be more precise what file needs editing (and where to see example how its supposed to be done / instructions).

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for the fixes.
About doc, please also update the `Supported Features` section in the board doc (`index.rst`)

File: boards/arm/stm32f411e_disco/doc/index.rst [hunk 3]
```diff
@@ -83,6 +83,8 @@ hardware features:
 +-----------+------------+-------------------------------------+
 | GPIO      | on-chip    | gpio                                |
 +-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                |
```
Comment (dbkinder) [inline boards/arm/stm32f411e_disco/doc/index.rst]: add a space to get that last column border to align on the right

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/stm32f411e_disco/doc/index.rst (hunk 3 above)
Comment (susundberg) [reply boards/arm/stm32f411e_disco/doc/index.rst]: Uh, sorry.

**Review by erwango — APPROVED** [review]

**Comment by susundberg** [comment]
Rewinded again to origin/master head -- is there anything still missing @dbkinder ?

**Review by dbkinder — APPROVED** [review]
+1 for doc changes, thanks.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**